selected/interface.go

```diff
@@ -0,0 +1,78 @@
+package selected
+
+import (
+	"fmt"
+)
+
+type Kind int
+
+func (k Kind) String() string {
+	switch k {
+	case FieldKind:
+		return "field"
+	case TypeAssertionKind:
+		return "type_assertion"
+	case TypenameFieldKind:
+		return "typename_field"
+	default:
+		panic(fmt.Errorf("invalid kind %d received", k))
+	}
+}
+
+const (
+	FieldKind Kind = iota
+	TypeAssertionKind
+	TypenameFieldKind
+)
+
+type Selection interface {
+	Kind() Kind
+}
+
+type Field interface {
+	Selection
+	Identifier() string
+	Aliased() string
+	Children() []Selection
+}
+
+type TypeAssertion interface {
+	Selection
+	Type() string
+	Children() []Selection
+}
+
+type TypenameField interface {
+	Selection
+	Type() string
+	Aliased() string
+}
+
+func Dump(selection Selection) {
```

Comment:
This is also present mainly for debugging/exploration purposes, another method that could be removed.